### PR TITLE
Feature/RR-2 validate add step

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**

### CODE REVIEW

File: server/validators/stepTargetDateValidator.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import moment from 'moment'
+
+export default function validateStepTargetDate(targetDate?: Date): Array<string> {
+  const errors: Array<string> = []
+
+  if (!targetDate) {
+    return errors
+  }
```
Comment (mattwills-moj-digital) [inline server/validators/stepTargetDateValidator.ts:8]: The `targetDate` is optional. Not sure if this is the best way to do it in typescript.

Comment (nathanrussell-moj-digital) [reply server/validators/stepTargetDateValidator.ts:8]: Thats OK, its basically saying if the targetDate value is "falsy" (null or undefined - they are 2 different values!). then return errors, which will be an empty array - so all good 👍

**Review by nathanrussell-moj-digital — DISMISSED** [review]
LGTM 👍 

Did you want me to look at the comment on my PR re: errors on the `-day` field first?

**Review by nathanrussell-moj-digital — APPROVED** [review]
ATOB 👍

**Pull request merged**